core/src/main/java/org/apache/iceberg/RemoveSnapshots.java

```diff
@@ -195,6 +299,18 @@ public void commit() {
     LOG.info("Committed snapshot changes");
 
     if (cleanExpiredFiles) {
+      TableMetadata updated = ops.refresh();
+      if (updated.refs() != null) {
```

Comment:
Why would `refs` be null?

Reply:
It cannot be, we're always maintaining the guarantee that  main will exist so there should be refs. Removed this check